packages/discord.js/src/structures/Guild.js

```diff
@@ -512,6 +513,15 @@ class Guild extends AnonymousGuild {
     }
   }
 
+  /**
+   * The URL to this guild's banner.
+   * @param {ImageURLOptions} [options={}] Options for the Image URL
+   * @returns {?string}
+   */
+  bannerURL({ format, size, dynamic } = {}) {
+    return this.banner && this.client.rest.cdn.Banner(this.id, this.banner, format, size, dynamic);
+  }
+
```

Comment:
This should instead be modified in the `AnonymousGuild` class instead:

https://github.com/discordjs/discord.js/blob/ed0cfd91edc3a2b23a34a8ecd9db38baa12b52fa/packages/discord.js/src/structures/AnonymousGuild.js#L69-L76

Both an `InviteGuild` and a `Guild` can receive the animated hash.